### PR TITLE
add upper bounds for vispy in napari

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Comes from https://github.com/napari/napari/issues/4891#issuecomment-1230417252

### CODE REVIEW

**Review by goanpeca — APPROVED** [review]

**Comment by dopplershift** [comment]
Can you post the output of `show_diff.py`?

**Review by jni — APPROVED** [review]
Awesome, thank you @jaimergp!

**Comment by jaimergp** [comment]
@dopplershift 

The `show_diff` output is part of the CI pipeline now. Pasting it here too:

```
2022-08-29T17:44:17.4298725Z + python show_diff.py
2022-08-29T17:44:18.0237455Z Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata_from_packages.json.bz2
2022-08-29T17:44:18.3564600Z Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata.json.bz2
2022-08-29T17:44:38.4102215Z noarch::napari-0.3.4-py_1.tar.bz2
2022-08-29T17:44:38.4104977Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4105986Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.4106794Z noarch::napari-0.3.6-py_0.tar.bz2
2022-08-29T17:44:38.4107804Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4108616Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.4120161Z noarch::napari-0.3.7-py_0.tar.bz2
2022-08-29T17:44:38.4121045Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4121570Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.4122214Z noarch::napari-0.3.8-py_0.tar.bz2
2022-08-29T17:44:38.4123143Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4123675Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.4124370Z noarch::napari-0.4.0-pyhd8ed1ab_0.tar.bz2
2022-08-29T17:44:38.4125030Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4125509Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.4126164Z noarch::napari-0.4.10-pyhd8ed1ab_0.tar.bz2
2022-08-29T17:44:38.4126835Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4127316Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.4127964Z noarch::napari-0.4.11-pyhd8ed1ab_0.tar.bz2
2022-08-29T17:44:38.4128642Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4129182Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.4129849Z noarch::napari-0.4.12-pyhd8ed1ab_0.tar.bz2
2022-08-29T17:44:38.4130875Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4131556Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.4132207Z noarch::napari-0.4.13-pyhd8ed1ab_0.tar.bz2
2022-08-29T17:44:38.4132985Z -    "vispy >=0.9.4",
2022-08-29T17:44:38.4133474Z +    "vispy >=0.9.4<0.10",
2022-08-29T17:44:38.4134478Z noarch::napari-0.4.13-pyhd8ed1ab_1.tar.bz2
2022-08-29T17:44:38.4135144Z -    "vispy >=0.9.4",
2022-08-29T17:44:38.4135587Z +    "vispy >=0.9.4<0.10",
2022-08-29T17:44:38.4136233Z noarch::napari-0.4.13-pyhd8ed1ab_2.tar.bz2
2022-08-29T17:44:38.4147354Z -    "vispy >=0.9.4",
2022-08-29T17:44:38.4148730Z +    "vispy >=0.9.4<0.10",
2022-08-29T17:44:38.4149646Z noarch::napari-0.4.14-pyhd8ed1ab_0.tar.bz2
2022-08-29T17:44:38.4150383Z -    "vispy >=0.9.4",
2022-08-29T17:44:38.4151045Z +    "vispy >=0.9.4<0.10",
2022-08-29T17:44:38.4151847Z noarch::napari-0.4.2-pyhd8ed1ab_0.tar.bz2
2022-08-29T17:44:38.4152513Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4152959Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.4153794Z noarch::napari-0.4.3-pyhd8ed1ab_0.tar.bz2
2022-08-29T17:44:38.4154839Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4156717Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.4159413Z noarch::napari-0.4.4-pyhd8ed1ab_0.tar.bz2
2022-08-29T17:44:38.4160383Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4160859Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.4161479Z noarch::napari-0.4.5-pyhd8ed1ab_0.tar.bz2
2022-08-29T17:44:38.4162131Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4162858Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.4163600Z noarch::napari-0.4.6-pyhd8ed1ab_0.tar.bz2
2022-08-29T17:44:38.4164242Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4164700Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.4165345Z noarch::napari-0.4.7-pyhd8ed1ab_0.tar.bz2
2022-08-29T17:44:38.4166004Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4166453Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.4167060Z noarch::napari-0.4.8-pyhd8ed1ab_0.tar.bz2
2022-08-29T17:44:38.4167704Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4168136Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.4168770Z noarch::napari-0.4.9-pyhd8ed1ab_0.tar.bz2
2022-08-29T17:44:38.4169405Z -    "vispy >=0.6.4",
2022-08-29T17:44:38.4169861Z +    "vispy >=0.6.4<0.10",
2022-08-29T17:44:38.6367496Z Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata_from_packages.json.bz2
2022-08-29T17:44:39.5057236Z Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata.json.bz2
2022-08-29T17:45:35.9267384Z Downloading: https://conda.anaconda.org/conda-forge/linux-armv7l/repodata_from_packages.json.bz2
2022-08-29T17:45:36.1020582Z Downloading: https://conda.anaconda.org/conda-forge/linux-armv7l/repodata.json.bz2
2022-08-29T17:45:36.8075639Z Downloading: https://conda.anaconda.org/conda-forge/linux-aarch64/repodata_from_packages.json.bz2
2022-08-29T17:45:37.1521209Z Downloading: https://conda.anaconda.org/conda-forge/linux-aarch64/repodata.json.bz2
2022-08-29T17:45:49.4085402Z Downloading: https://conda.anaconda.org/conda-forge/linux-ppc64le/repodata_from_packages.json.bz2
2022-08-29T17:45:50.2470097Z Downloading: https://conda.anaconda.org/conda-forge/linux-ppc64le/repodata.json.bz2
2022-08-29T17:46:01.5999776Z Downloading: https://conda.anaconda.org/conda-forge/osx-64/repodata_from_packages.json.bz2
2022-08-29T17:46:02.6469081Z Downloading: https://conda.anaconda.org/conda-forge/osx-64/repodata.json.bz2
2022-08-29T17:47:01.8920227Z Downloading: https://conda.anaconda.org/conda-forge/osx-arm64/repodata_from_packages.json.bz2
2022-08-29T17:47:02.2515647Z Downloading: https://conda.anaconda.org/conda-forge/osx-arm64/repodata.json.bz2
2022-08-29T17:47:10.3986363Z Downloading: https://conda.anaconda.org/conda-forge/win-32/repodata_from_packages.json.bz2
2022-08-29T17:47:10.8051805Z Downloading: https://conda.anaconda.org/conda-forge/win-32/repodata.json.bz2
2022-08-29T17:47:17.4046405Z Downloading: https://conda.anaconda.org/conda-forge/win-64/repodata_from_packages.json.bz2
2022-08-29T17:47:17.8332324Z Downloading: https://conda.anaconda.org/conda-forge/win-64/repodata.json.bz2
2022-08-29T17:47:59.6815931Z + exit 0
```

**Comment by jaimergp** [comment]
@dopplershift Can you take another look? Thanks!

**Comment by psobolewskiPhD** [comment]
Bumping this, as it's still a problem on the napari side.

**Comment by hmaarrfk** [comment]
hmm, fun.

**Comment by psobolewskiPhD** [comment]
Should there be a delimiter? Like:
`"vispy >=0.6.4, <0.10"`

File: recipe/gen_patch_json.py [hunk 1]
```diff
@@ -1500,8 +1500,13 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
...
         # pillow 7.1.0 and 7.1.1 break napari viewer but this wasn't dealt with til latest release
-        if record_name == "napari" and record.get("timestamp", 0) < 1642529454000:  # 2022-01-18
-            _replace_pin("pillow", "pillow !=7.1.0,!=7.1.1", record.get("depends", []), record)
+        if record_name == "napari":
+            timestamp = record.get("timestamp", 0) 
+            if timestamp < 1642529454000:  # 2022-01-18
+                _replace_pin("pillow", "pillow !=7.1.0,!=7.1.1", record.get("depends", []), record)
+            if timestamp < 1661793597230:  # 2022-08-29
+                _replace_pin("vispy >=0.9.4", "vispy >=0.9.4<0.10", record.get("depends", []), record)
+                _replace_pin("vispy >=0.6.4", "vispy >=0.6.4<0.10", record.get("depends", []), record)
```
Comment (jakirkham) [inline recipe/gen_patch_json.py:1509]: Agreed. It should be...

```suggestion
                _replace_pin("vispy >=0.9.4", "vispy >=0.9.4,<0.10", record.get("depends", []), record)
                _replace_pin("vispy >=0.6.4", "vispy >=0.6.4,<0.10", record.get("depends", []), record)
```

Comment (jakirkham) [reply recipe/gen_patch_json.py:1509]: See PR ( https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/333 )

Comment (psobolewskiPhD) [reply recipe/gen_patch_json.py:1509]: Just made a PR with that:
https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/333

**Pull request merged**